spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/ExpireSnapshotsProcedure.java

```diff
@@ -102,6 +104,10 @@ public StructType outputType() {
         action.executeDeleteWith(executorService(maxConcurrentDeletes, "expire-snapshots"));
       }
 
+      if (streamResult) {
+        action.option("stream-results", streamResult.toString());
```

Comment:
@aokolnychyi, why not use this by default?

Reply:
There is one [PR](https://github.com/apache/iceberg/pull/1395) previously to collect all the results by default. This should be most usual cases. 
The `stream-results` option should be explicitly enabled for the case that contains huge amount of data files.

Reply:
Would it be possible / desirable to just pass `streamResults` as is (either true or false), to avoid the conditional check? It should be an accepted option for the action either way.

Reply:
@rdblue, in most cases, calling `toLocalIterator` would be much more expensive as it would trigger multiple jobs (as far as I remember). We started with that but then changed in the PR that @0xffmeta linked.

@RussellSpitzer may have more context.

Reply:
@kbendick just updated the PR to remove the conditional check.

Reply:
toLocalIterator is much more expensive since every Spark task becomes a full Spark Job.

Reply:
The reason for #1395 if anyone remembers is that having toLocalIterator on made this test suite take on the order of ten's of minutes.  I'm not sure what happened here though

https://github.com/apache/iceberg/pull/1395/files#diff-c514ec43081677e77a31b19f98b424734da476b74043733d79f37c924b5dad5eR1031-R1032

As the test that was renamed to "testCollectAsList" was actually using the local iterator before